tests/purs/failing/3549.out

```diff
@@ -3,15 +3,21 @@ in module [33mMain[0m
 at tests/purs/failing/3549.purs:8:78 - 8:79 (line 8, column 78 - line 8, column 79)
 
   Could not match kind
-
-    [33mType[0m
-
+  [33m              [0m
+  [33m  Type -> Type[0m
+  [33m              [0m
   with kind
-
-    [33mType -> Type[0m
-
-
-while checking the kind of [33mFunctor f => (a -> b) -> f a -> f b[0m
+  [33m      [0m
+  [33m  Type[0m
+  [33m      [0m
+
+while checking that type [33mf[0m
+  has kind [33mType -> Type[0m
+while inferring the kind of [33mFunctor f[0m
```

Comment:
I don't know what we would filter on. The issue is that foralls are inductive, which requires an inference step for each type variable.

Reply:
Ok then - if it's not immediately clear how to do this I'm happy to not worry about this for now.

Reply:
The way to minimize would be in the kind checker by not checking foralls inductively if possible. This would require collecting the flat forall list, splitting them into dependency groups (`a` needs to be checked before `b` because the kind of `a` depends on `b`), and work inductively on those groups instead of each binding.